harness/determined/profiler.py

```diff
@@ -0,0 +1,559 @@
+                    pass
+
+            next_collection += self.MEASUREMENT_INTERVAL
+
+            # One-time initialization
+            if last_measurement_time is None:
+                last_measurement_time = time.time()
+                network_throughput_collector.reset()
+                disk_collector.reset()
+
+            # Check if it is time to take a new measurement
+            if time.time() - last_measurement_time > self.MEASUREMENT_INTERVAL:
+                immutable_batch_idx = self.current_batch_idx
+                cpu_util = cpu_util_collector.measure(immutable_batch_idx)
+                gpu_util = gpu_util_collector.measure(immutable_batch_idx)
+                gpu_memory = gpu_memory_collection.measure(immutable_batch_idx)
+                net_thru_sent, net_thru_recv = network_throughput_collector.measure(
+                    immutable_batch_idx
+                )
+                free_memory = free_memory_collector.measure(immutable_batch_idx)
+                disk_read_thru, disk_write_thru, iops = disk_collector.measure(immutable_batch_idx)
+
+                for gpu_uuid in gpu_util.keys():
+                    self.current_batch.add_gpu_measurement(
+                        SysMetricType.GPU_UTIL_METRIC, gpu_uuid, gpu_util[gpu_uuid]
+                    )
+
+                for gpu_uuid in gpu_memory.keys():
+                    self.current_batch.add_gpu_measurement(
+                        SysMetricType.GPU_FREE_MEMORY_METRIC, gpu_uuid, gpu_util[gpu_uuid]
+                    )
+
+                self.current_batch.add_nongpu_measurement(
+                    SysMetricType.NET_THRU_SENT_METRIC, net_thru_sent
+                )
+                self.current_batch.add_nongpu_measurement(
+                    SysMetricType.NET_THRU_RECV_METRIC, net_thru_recv
+                )
+                self.current_batch.add_nongpu_measurement(SysMetricType.DISK_IOPS_METRIC, iops)
+                self.current_batch.add_nongpu_measurement(
+                    SysMetricType.DISK_THRU_READ_METRIC, disk_read_thru
+                )
+                self.current_batch.add_nongpu_measurement(
+                    SysMetricType.DISK_THRU_WRITE_METRIC, disk_write_thru
+                )
+                self.current_batch.add_nongpu_measurement(
+                    SysMetricType.FREE_MEM_METRIC, free_memory
+                )
+                self.current_batch.add_nongpu_measurement(
+                    SysMetricType.SIMPLE_CPU_UTIL_METRIC, cpu_util
+                )
+                last_measurement_time = time.time()
+
+            # Check if it is time to flush the batch and start a new batch
+            if time.time() - batch_start_time > self.FLUSH_INTERVAL:
+                self.send_queue.put(self.current_batch.convert_to_post_format())
+                self.current_batch.clear()
+                batch_start_time = time.time()
+
+
+# This is a thread that exists solely so that we can make API calls without blocking
```

Comment:
nit: can you move this to the docstring of this class.

Reply:
Fixed